lib/dynamoid/undumping.rb

```diff
@@ -202,7 +202,7 @@ def process(value)
                             else
                               @options[:store_as_string]
                             end
-        value = DateTime.iso8601(value).to_time.to_i if use_string_format
+        value = DateTime.iso8601(value).to_time if use_string_format
```

Comment:
I believe it should be tested as well. It's a new feature - storing milliseconds when use string format for datetime.

Reply:
I agree, I think this could be a clearer abstraction. FWIW, This was amended to make the current and additional spec coverage pass